Sources/Twift.swift

```diff
@@ -70,9 +71,11 @@ public class Twift: NSObject, ObservableObject {
     return encoder
   }
   
-  /// Refreshes the OAuth 2.0 token, optionally forcing a refresh even if the token is still valid
-  /// - Parameter onlyIfExpired: Set to false to force the token to refresh even if it hasn't yet expired.
-  public func refreshOAuth2AccessToken(onlyIfExpired: Bool = true) async throws {
+	/// Refreshes the OAuth 2.0 token, optionally forcing a refresh even if the token is still valid
+	/// - Parameters:
+		/// - onlyIfExpired: Set to false to force the token to refresh even if it hasn't yet expired.
+		/// - onRefresh: An optional callback method to be called after successful token refresh.
+    public func refreshOAuth2AccessToken(onlyIfExpired: Bool = true, onRefresh: ((OAuth2User)-> Void)? = nil) async throws {
```

Comment:
The `onRefresh` callback should go in the initialiser for `Twift` instances and get called in this method. That way, you don’t have to manually check the token expiration and refresh it; you can rely on Twift’s automatic token refreshing and just provide your own handler when initialising a `Twift` instance for how new tokens are handled.

Reply:
Oh wow, that is 🤌
Never would have thought to have that kind of implementation. I think I get ya, will revise soon.